src/applets/budgie-menu/views/ListView.vala

```diff
@@ -0,0 +1,483 @@
+
+		this.content_scroll = new Gtk.ScrolledWindow(null, null) {
+			overlay_scrolling = true,
+			hscrollbar_policy = Gtk.PolicyType.NEVER,
+			vscrollbar_policy = Gtk.PolicyType.AUTOMATIC,
+			min_content_height = HEIGHT
+		};
+		this.content_scroll.set_overlay_scrolling(true);
+		this.content_scroll.add(applications);
+		right_layout.pack_start(content_scroll, true, true, 0);
+
+		// placeholder in case of no results
+		var placeholder = new Gtk.Label("<big>%s</big>".printf(_("Sorry, no items found"))) {
+			use_markup = true,
+			margin = 6,
+		};
+		placeholder.get_style_context().add_class("dim-label");
+		placeholder.show();
+		this.applications.set_placeholder(placeholder);
+
+		this.settings.changed.connect(on_settings_changed);
+		this.on_settings_changed("menu-compact");
+		this.on_settings_changed("menu-headers");
+		this.on_settings_changed("menu-categories-hover");
+		this.on_settings_changed("menu-show-control-center-items");
+
+		// management of our listbox
+		this.applications.set_filter_func(do_filter_list);
+		this.applications.set_sort_func(do_sort_list);
+	}
+
+	/**
+	 * Refreshes the category and application lists.
+	 */
+	public override void refresh(Tracker app_tracker) {
+		lock (this.reloading) {
+			if (this.reloading) {
+				return;
+			}
+			this.reloading = true;
+		}
+
+		// Destroy all application items
+		foreach (var child in this.applications.get_children()) {
+			child.destroy();
+		}
+		this.application_buttons.remove_all();
+		this.control_center_buttons.clear();
+
+		// Destroy all category items
+		foreach (var child in this.categories.get_children()) {
+			child.destroy();
+		}
+
+		// Load all of the new content in the background
+		Idle.add(() => {
+			this.load_menus(app_tracker);
+			this.invalidate();
+			return false;
+		});
+
+		lock (this.reloading) {
+			this.reloading = false;
+		}
+	}
+
+	/**
+	 * Build the category and application lists.
+	 */
+	private void load_menus(Tracker app_tracker) {
+		// "All" button"
+		this.all_categories = new CategoryButton(null);
+		this.all_categories.enter_notify_event.connect(this.on_mouse_enter);
+		this.all_categories.toggled.connect(()=> {
+			this.update_category(this.all_categories);
+		});
+		this.categories.pack_start(this.all_categories, false, false, 0);
+
+		foreach (var category in app_tracker.categories) {
+			// Skip empty categories
+			if (category.apps.is_empty) {
+				continue;
+			}
+
+			// Create a new button for this category
+			var btn = new CategoryButton(category);
+			btn.join_group(all_categories);
+			btn.enter_notify_event.connect(this.on_mouse_enter);
+
+			// Ensures we find the correct button
+			btn.toggled.connect(() => {
+				this.update_category(btn);
+			});
+
+			btn.show_all();
+			this.categories.pack_start(btn, false, false, 0); // Add the button
+
+			// Create a button for each app in this category
+			foreach (var app in category.apps) {
+				var app_btn = new MenuButton(app, category, icon_size);
```

Comment:
IMO the logic should be here instead of Tracker for determining if we should show the control panels and skip if the app is a panel and the setting isn't enabled.